core/token_manager.go

```diff
@@ -48,27 +49,42 @@ type TokenManager struct {
 	userAccessToken string
 	iamAPIkey       string
 	iamURL          string
+	iamClientId     string
+	iamClientSecret string
 	tokenInfo       *TokenInfo
 	client          *http.Client
 }
 
 // NewTokenManager : Instantiate TokenManager
-func NewTokenManager(iamAPIkey string, iamURL string, userAccessToken string) *TokenManager {
+func NewTokenManager(iamAPIkey string, iamURL string, userAccessToken string,
+    iamClientId string, iamClientSecret string) (*TokenManager, error) {
 	if iamURL == "" {
 		iamURL = DefaultIAMURL
 	}
+	
+	if iamClientId == "" && iamClientSecret == "" {
+	    iamClientId = DefaultIAMClientId
+	    iamClientSecret = DefaultIAMClientSecret
+	} else if iamClientId != "" && iamClientSecret != "" {
```

Comment:
Do we need this?

Reply:
yes because if neither of those values are specified, then we want to use the default values (bx/bx) for basic auth.
If both are specified, then we want to use them.
If only 1 is specified, then it's an error... both the id and secret need to be specified together or neither of them should be specified.